pymc/distributions/distribution.py

```diff
@@ -397,6 +398,15 @@ def __new__(
         if not isinstance(name, string_types):
             raise TypeError(f"Name needs to be a string but got: {name}")
 
+        if transform is None and default_transform is UNSET:
```

Comment:
This warning can be in the relevant section in `pm.Model` instead of `Distribution`